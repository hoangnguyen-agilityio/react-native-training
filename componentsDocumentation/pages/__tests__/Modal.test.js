import React from 'react';
import renderer from 'react-test-renderer';
import ModalPage from '../Modal';
import {Modal, Pressable} from 'react-native';

test('renders correctly', () => {
  const tree = renderer.create(<ModalPage />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('page elements', () => {
  const tree = renderer.create(<ModalPage />);
  const modal = tree.root.findAllByType(Modal);

  expect(modal).toHaveLength(1);
});

test('Show / hide modal when click button', () => {
  const tree = renderer.create(<ModalPage />);
  const modal = tree.root.findAllByType(Modal);

  tree.root.findAllByType(Pressable)[0].props.onPress();

  expect(modal[0].props.visible).toBeTruthy();

  tree.root.findAllByType(Pressable)[0].props.onPress();

  expect(modal[0].props.visible).toBeFalsy();
});
