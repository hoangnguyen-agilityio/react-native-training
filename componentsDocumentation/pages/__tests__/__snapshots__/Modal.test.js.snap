// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 22,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 20,
          "elevation": 2,
          "padding": 10,
        },
        Object {
          "backgroundColor": "#F194FF",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Show Modal
    </Text>
  </View>
</View>
`;
